### PR TITLE
New compiled FST implementation

### PR DESCRIPTION
**Pull request by cmoen**
Here's a new compiled FST implementation that Fujinuma-san and myself have been working on over the weekend.

### CODE REVIEW

File: kuromoji-core/src/main/java/com/atilika/kuromoji/fst/State.java [hunk 1]
```diff
@@ -10,6 +26,8 @@
     boolean visited; //for visualization purpose
     private int targetJumpAddress = -1;
```
Comment (akkikiki) [inline kuromoji-core/src/main/java/com/atilika/kuromoji/fst/State.java]: This is used only for the old FSTCompiler so perhaps you can remove this to (and refactor it).

File: kuromoji-core/src/main/java/com/atilika/kuromoji/fst/State.java [hunk 2]
```diff
@@ -22,6 +40,14 @@ public State(State source) {
...
 
+    public int getNewTargetAddress() {
+        return newTargetAddress;
+    }
+
+    public void setNewTargetAddress(int newTargetAddress) {
+        this.newTargetAddress = newTargetAddress;
+    }
+
     public int getTargetJumpAddress() {
```
Comment (akkikiki) [inline kuromoji-core/src/main/java/com/atilika/kuromoji/fst/State.java]: Same as above.

**Comment by akkikiki** [comment]
Great work! Looks good to me!

**Comment by cmoen** [comment]
Thanks, Fujinuma-san.  I'm merging this.

**Pull request merged**